src/repositories/media.repository.ts

```diff
@@ -168,4 +168,19 @@ export class MediaRepository extends Repository {
     });
     return body;
   }
+
+  public async getInsights(id: string) {
+    const { body } = await this.client.request.send({
+      url: '/api/v1/ads/graphql/?locale=en_US&vc_policy=insights_policy&surface=post',
```

Comment:
This request can not be placed in this repository. The right place for this - `ads.repository.ts`

Reply:
But, it has nothing to do with ads?

Reply:
So, why instagram implemented this endpoint with `ads` domain?

Reply:
Are you placing requests based on instagram routes or by logical association?

Reply:
As you could notice - i'm placing requests based on instagram routes 😄 just watch any other repository

Reply:
I hope this 
https://github.com/dilame/instagram-private-api/blob/master/src/repositories/README.md
Will be helpfull

Reply:
Sorry about this. Will fix it right away. 

Response to this request is a bit weird. Do you want me to generate response class or to write something that makes a little bit more sense?

Reply:
Based on the previous experience - it is better to not significantly modify original responses because instagram changes it too often. The purpose of this library is just be a convenient middleware between instagram api and user-land code. So generate response class will be just enough.
BTW, this package also provides higher-level API, based on logical association, we place it in `entities` and `services`. Those API's provides `like-in-app` actions, and must be named respectively like it named in the official IG app.

Reply:
In your case it will be ideal to create instance method in `media.entity.ts` that will call `ads.graphql()` under the hood.

Reply:
Got it. Thanks